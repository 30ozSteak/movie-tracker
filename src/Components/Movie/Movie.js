import React, { Component } from 'react';
import * as api from '../../Helpers/apiCaller.js';
import { connect } from 'react-redux';
import { addFavThunk } from '../../Thunks/addFav';
import { favsLocalThunk } from '../../Thunks/favsLocal.js';




class Movie extends Component {
  constructor() {
    super();
  }

  toggleFav = () => {
    const { toggleFav, movie, userID, resetFavs } = this.props
    toggleFav(movie, userID)
    resetFavs()

  }

  render() {

    const imgURL = 'https://image.tmdb.org/t/p/original';

    const { movie } = this.props;

    return (

      <article className="movie-card">
        <section className='img-box'>
          <div className="fave-star" onClick={this.toggleFav} />
          <img className="movie-img" src={`${imgURL}${movie.poster}`} />
        </section>
        <h3 className="movie-title">{movie.title.toLowerCase()}</h3>
        <section className="date-box">
          <p className="movie-overview">{movie.overview.toLowerCase()}</p>
        </section>
      </article>

    )
  }
}

export const mapStateToProps = state => ({
  userID: state.user.id
})

export const mapDispatchToProps = dispatch => ({
  toggleFav: (movie, userID) => dispatch(addFavThunk(movie, userID))

})
export default connect(mapStateToProps, mapDispatchToProps)(Movie);

Movie.propTypes = {
  toggleFav: PropTypes.func,
  movie: PropTypes.object,
  movieID: PropTypes.number,
  userID: propTypes.number,
  resetFavs: propTypes.func
}