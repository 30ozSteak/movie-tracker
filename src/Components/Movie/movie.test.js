/* eslint-disable */


import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, mount} from 'enzyme';
import Movie from './Movie';


describe('Movie', ()=>{

  let wrapper;
  let props;

  beforeEach(()=>{

    let props = {
      movie: [{movie: 'movie1'}]
    }

    wrapper = shallow(<Movie movie={props}/>);

  })

  it('should match the snapshot', ()=>{
    expect(wrapper).toMatchSnapshot();
  })
})
