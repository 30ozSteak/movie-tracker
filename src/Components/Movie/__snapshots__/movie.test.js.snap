// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie
    movie={
      Object {
        "movie": Array [
          Object {
            "movie": "movie1",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Title: 
        </h3>,
        <time>
          Release date: 
        </time>,
        <p>
          Overview: 
        </p>,
        <img
          src="https://image.tmdb.org/t/p/w185_and_h278_bestv2undefined"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Title: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Title: ",
          undefined,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Release date: ",
            undefined,
          ],
          "dateTime": undefined,
        },
        "ref": null,
        "rendered": Array [
          "Release date: ",
          undefined,
        ],
        "type": "time",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Overview: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Overview: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "https://image.tmdb.org/t/p/w185_and_h278_bestv2undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Title: 
          </h3>,
          <time>
            Release date: 
          </time>,
          <p>
            Overview: 
          </p>,
          <img
            src="https://image.tmdb.org/t/p/w185_and_h278_bestv2undefined"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Title: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Title: ",
            undefined,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Release date: ",
              undefined,
            ],
            "dateTime": undefined,
          },
          "ref": null,
          "rendered": Array [
            "Release date: ",
            undefined,
          ],
          "type": "time",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Overview: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Overview: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "https://image.tmdb.org/t/p/w185_and_h278_bestv2undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
